connectivity/FEATURE_BLE/source/common/ble_trace_helpers.h

```diff
@@ -45,9 +45,47 @@ static inline const char* tr_as_array(T item)
     return (mbed_trace_array)((const uint8_t*)&item, sizeof(item));
 }
 
-static inline const char* to_string(UUID uuid)
+static inline char* to_string(const UUID& uuid)
 {
-    return (mbed_trace_array)(uuid.getBaseUUID(), uuid.getLen());
+    const uint8_t* Buffer    = uuid.getBaseUUID();
+    const uint8_t  Length    = uuid.getLen();
+    const  size_t  Row_Count = 4;
+
+    const char *Hex = "0123456789ABCDEF";
+
+    static char string_buffer[Row_Count][UUID::LENGTH_OF_LONG_UUID + /* Number of hyphen delimiters =*/ 4];
+    static uint8_t idx = 0;
+
+    if (idx == Row_Count) {
+        idx= 0;
+    } else {
+        idx++;
```

Comment:
That's not correct, it should be increment then check if it is out of range. At the moment, it will access `string_buffer[4]` which is out of range.